src/Nethermind/Nethermind.Network/Discovery/Messages/TopicRegisterMessage.cs

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2018 Demerzel Solutions Limited
+ * This file is part of the Nethermind library.
+ *
+ * The Nethermind library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Nethermind library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+using Nethermind.Core.Extensions;
+using Nethermind.Core.Crypto;
+
+namespace Nethermind.Network.Discovery.Messages
+{
+    public class TopicRegisterMessage : DiscoveryMessage
+    {
+        public Topic[] Topics { get; set; }
+
+        public short Idx { get; set; }
+
+        public PongMessage Pong { get; set; }
```

Comment:
a message nested in a message - was that desired? probably as I go through messages I will see some reasoning behind it